sqlmesh/core/snapshot/definition.py

```diff
@@ -219,22 +225,39 @@ def catalog_for_environment(
     ) -> t.Optional[str]:
         return environment_naming_info.catalog_name_override or self.catalog
 
-    def schema_for_environment(self, environment_naming_info: EnvironmentNamingInfo) -> str:
+    def schema_for_environment(
+        self, environment_naming_info: EnvironmentNamingInfo, dialect: DialectType = None
+    ) -> str:
         schema = self.schema_name or c.DEFAULT_SCHEMA
         if (
             environment_naming_info.name.lower() != c.PROD
             and environment_naming_info.suffix_target.is_schema
         ):
-            schema = f"{schema}__{environment_naming_info.name}"
+            env_name = environment_naming_info.name
+            if environment_naming_info.normalize_name:
+                env_name = normalize_identifiers(env_name, dialect=dialect).name
+                if not self.schema_name:
+                    # We also normalize the schema if it's the default one
```

Comment:
Why weren't we doing this before? A bug?

Reply:
Yeah, I think so. If we'd set the schema earlier in the control flow, we wouldn't have this issue. But until now, we were basically interpolating a lowercase `default` schema if none was specified, i.e. `"default"."FOO"` would be created in SF.